EpikCord/__init__.py

```diff
@@ -1156,6 +1113,151 @@ async def command_error(
     ):
         logger.exception(error)
 
+    #AutoModeration Events
+    async def auto_moderation_rule_create(self, data:dict) -> AutoModerationRule:
+        rule = AutoModerationRule(self, data)
+        self.auto_moderation_cache.add_to_cache(rule.guild_id, rule)
+        return rule
+    
+    async def auto_moderation_rule_update(self, data:dict)->tuple[Optional[AutoModerationRule], AutoModerationRule]:
+        after:AutoModerationRule = AutoModerationRule(self,data)
+        before:Optional[AutoModerationRule] = self.auto_moderation_cache.get(after.guild_id)
+        return before,after
+        
+    async def auto_moderation_rule_delete(self, data:dict)->Optional[AutoModerationRule]:
+        rule_guild_id = data["guild_id"]
+        previous:AutoModerationRule = self.auto_moderation_cache.get(rule_guild_id)
+        return previous
+
+    async def auto_moderation_action_execution(self, data:dict) -> AutoModerationExecutionEvent:
+        return AutoModerationExecutionEvent(data)    
+
+    # Guild Events
+    async def guild_create(self, data) -> Union[UnavailableGuild, Guild]:
+        guild = (
+            UnavailableGuild(data)
+            if data.get("unavailable") is True
+            else Guild(self, data)
+            if data.get("unavailable") is False
+            else None
+        )
+
+        if not guild:
+            return
+
+        self.guilds.add_to_cache(guild.id, guild)
+
+        if data.get("unavailable") is None:
+            return  # Bot was removed
+
+        for channel in data["channels"]:
+            self.channels.add_to_cache(
+                data["id"], self.utils.channel_from_type(channel)
+            )
+
+        for thread in data["threads"]:
+            self.channels.add_to_cache(data["id"], self.utils.channel_from_type(thread))
+
+        return guild
+
+        # TODO: Add other attributes to cache
+    async def guild_update(self, data) -> tuple[Optional[Guild], Guild]:
+        after:Guild = Guild(self, data)
+        before:Optional[Guild] = self.guilds.fetch(after.id)
+        return before ,after
+    
+    async def guild_delete(self, data: dict) -> Optional[Guild]:
+        return self.guilds.get(data["id"])
+
+    async def guild_ban_add(self, data:dict) -> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self, data["user"])
```

Comment:
Because the event will never return invalid data, the guild_id will always be present. You should use cache, and *then* fetch.